project/src/com/google/daggerquery/dagger_query_textproto.bzl

```diff
@@ -0,0 +1,28 @@
+def dagger_query_textproto(name, dagger_app_target):
+    _dagger_query_textproto(name = name, dagger_app_target = dagger_app_target)
+
+def _dagger_query_textproto_impl(ctx):
+    dagger_app_target = ctx.attr.dagger_app_target
+    src_jars = dagger_app_target[JavaInfo].source_jars
+
+    ctx.actions.run_shell(
+        inputs = src_jars,
+        outputs = [ctx.outputs.out],
+        command = "unzip -p -o {src_jar} binding_graph.textproto > {out}".format(
+            src_jar = src_jars[0].path,
```

Comment:
Sorry Rita I ran into problems setting this up internally so I wasn't able to debug this. I think what you can do is to print out the path of this `src_jar`, and manually examine the contents of this jar to ensure the `binding_graph.textproto` file is actually there. This will help narrow down whether it's a problem with the shell command, we're feeding in the wrong jar, or there was a problem packaging the textproto file into the jar.

Reply:
No problem, it's good idea, thanks!
I will try to reproduce these steps 👌